pysteps/nowcasts/anvil.py

```diff
@@ -0,0 +1,370 @@
+    if measure_time:
+        init_time = time.time() - starttime_init
+
+    print("Starting nowcast computation.")
+
+    if measure_time:
+        starttime_mainloop = time.time()
+
+    r_f = []
+    dp = None
+    for t in range(n_timesteps):
+        print("Computing nowcast for time step %d... " % (t + 1), end="")
+        sys.stdout.flush()
+        
+        if measure_time:
+            starttime = time.time()
+
+        for i in range(n_cascade_levels):
+            vil_dec[i, :] = autoregression.iterate_ar_model(vil_dec[i, :], phi[i])
+
+        vil_dec_dict = {}
+        vil_dec_dict["cascade_levels"] = vil_dec[:, -1, :]
+        vil_dec_dict["domain"] = "spatial"
+        vil_dec_dict["normalized"] = False
+        vil_f = recomp_method(vil_dec_dict)
+        vil_f[~mask] = np.nan
+
+        if rainrate is not None:
+            r_f_ = r_vil_a * vil_f + r_vil_b
+        else:
+            r_f_ = vil_f
+
+        extrap_kwargs.update({"D_prev": dp, "return_displacement": True,
+                              "allow_nonfinite_values": True})
+        r_f_, dp = extrapolator(r_f_, velocity, 1, **extrap_kwargs)
+
+        if measure_time:
+            print("%.2f seconds." % (time.time() - starttime))
+        else:
+            print("done.")
+
+        r_f.append(r_f_[-1])
+
+    if measure_time:
+        mainloop_time = time.time() - starttime_mainloop
+
+    if measure_time:
+        return np.stack(r_f), init_time, mainloop_time
+    else:
+        return np.stack(r_f)
+
+
+def _estimate_ar1_params(gamma):
+    phi = []
+    phi.append(1 + gamma[0, :])
+    phi.append(-gamma[0, :])
+    phi.append(np.zeros(gamma[0, :].shape))
+
+    return phi
+
+
+def _estimate_ar2_params(gamma):
+    phi_diff = []
+    phi_diff.append(gamma[0, :] * (1 - gamma[1, :]) / (1 - gamma[0, :]*gamma[0, :]))
+    phi_diff.append((gamma[1, :] - gamma[0, :]*gamma[0, :]) / (1 - gamma[0, :]*gamma[0, :]))
+
+    phi = []
+    phi.append(1 + phi_diff[0])
+    phi.append(-phi_diff[0] + phi_diff[1])
+    phi.append(-phi_diff[1])
+    phi.append(np.zeros(phi_diff[0].shape))
+
+    return phi
+
+
+def _moving_window_corrcoef(x, y, window_radius):
+    mask = np.logical_and(np.isfinite(x), np.isfinite(y))
+    x = x.copy()
+    x[~mask] = 0.0
+    y = y.copy()
+    y[~mask] = 0.0
+    mask = mask.astype(float)
+
+    n = gaussian_filter(mask, window_radius, mode="constant")
+
+    ssx = gaussian_filter(x**2, window_radius, mode="constant")
+    ssy = gaussian_filter(y**2, window_radius, mode="constant")
+    sxy = gaussian_filter(x*y, window_radius, mode="constant")
+
+    stdx = np.sqrt(ssx / n)
+    stdy = np.sqrt(ssy / n)
+    cov = sxy / n
+
+    mask = np.logical_and(stdx > 1e-8, stdy > 1e-8)
+    mask = np.logical_and(mask, stdx * stdy > 1e-8)
+    mask = np.logical_and(mask, n > 1e-3)
+    corr = np.empty(x.shape)
+    corr[mask] = cov[mask] / (stdx[mask] * stdy[mask])
+    corr[~mask] = 0.0
+
+    return corr
+
+
+def _R_VIL_regression(vil, r, window_radius):
```

Comment:
I think it's good to put some comments in this function. Perhaps good to refer to parts of your paper here! :) Same for line 157 were this function is used.